### PR TITLE
fgets and utf8 functions for Windows 

### PR DESCRIPTION
**Pull request by manojampalam**

### CODE REVIEW

File: sftp.c [hunk 1]
```diff
@@ -2211,54 +2158,22 @@ interactive_loop(struct sftp_conn *conn, char *file1, char *file2)
...
-					break;
-				}
-				else {
-					char *pcmd = NULL;
-					if ((pcmd = utf16_to_utf8(wcmd)) == NULL)
-						fatal("failed to convert input arguments");
-					strcpy(cmd, pcmd);
-					free(pcmd);                    
-				}
-			} else if (fgets(cmd, sizeof(cmd), infile) == NULL)
```
Comment (bagajjal) [inline sftp.c:2251]: Please check this.. last time it didn't work when I try to read using fgetws()
https://github.com/PowerShell/Win32-OpenSSH/issues/529

File: contrib/win32/win32compat/win32-utf8.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+int
+mprintf(const char *fmt, ...) 
+{
+	va_list valist;
+	va_start(valist, fmt);
+	vfmprintf(stdout, fmt, valist);
+	va_end(valist);
+	return 0;
```
Comment (bagajjal) [inline contrib/win32/win32compat/win32-utf8.c]: return the return value of vfmprintf.. it is useful when we want to check the number of characters written...

File: contrib/win32/win32compat/win32-utf8.c [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+int
+snmprintf(char *buf, size_t len, int *written, const char *fmt, ...) 
+{
+	int num;
+	va_list valist;
+	va_start(valist, fmt);
+	num = vsnprintf(buf, len, fmt, valist);
+	va_end(valist);
+	*written = num;
+	return 0;
```
Comment (bagajjal) [inline contrib/win32/win32compat/win32-utf8.c:44]: return the return value of vfmprintf.. it is useful when we want to check the number of characters written...

**Review by bagajjal — CHANGES_REQUESTED** [review]

File: sftp.c (hunk 1 above)
Comment (manojampalam) [reply sftp.c:2251]: If you look into implementation of w32_fgets, it only calls fgetws for console io, file io is still on fgets()

**Pull request merged**